specs/online_order_spec.rb

```diff
@@ -3,69 +3,137 @@
+		id = 1
+		products = {'socks' => 2.34}
+		customer = 5
+		status = 'paid'
+		online_order = Grocery::OnlineOrder.new(id, products, customer, status)
+		
+		calculated_total = online_order.total
+		raw_total = (2.34*1.075).round(2) + 10
+		
+		calculated_total.must_equal raw_total
     end
 
     it "Doesn't add a shipping fee if there are no products" do
-      # TODO: Your test code here!
+		id = 1
+		products = {}
+		customer = 5
+		status = 'pending'
+		online_order = Grocery::OnlineOrder.new(id, products, customer, status)
+		
+		total = online_order.total
+		
+		total.must_equal 0
     end
   end
 
   describe "#add_product" do
     it "Does not permit action for processing, shipped or completed statuses" do
-      # TODO: Your test code here!
+		id = 1
+		products = {'socks' => 2.34}
+		customer = 5
+		status = 'paid'
+		online_order = Grocery::OnlineOrder.new(id, products, customer, status)
     end
 
     it "Permits action for pending and paid satuses" do
-      # TODO: Your test code here!
+		id = 1
+		products = {'socks' => 2.34}
+		customer = 5
+		status = 'paid'
+		online_order = Grocery::OnlineOrder.new(id, products, customer, status)
+		count_before = online_order.products.keys.count + 1
+		online_order.add_product('shoes', 3)
+		count_after = online_order.products.keys.count
+		
+		count_after.must_equal count_before
     end
   end
 
   describe "OnlineOrder.all" do
     it "Returns an array of all online orders" do
-      # TODO: Your test code here!
-      # Useful checks might include:
-      #   - OnlineOrder.all returns an array
-      #   - Everything in the array is an Order
-      #   - The number of orders is correct
-      #   - The customer is present
-      #   - The status is present
-      # Feel free to split this into multiple tests if needed
-    end
+		online_orders = Grocery::OnlineOrder.all
+		
+		online_orders.must_be_kind_of Array
+		
+		online_orders.each { |i|
+			i.must_be_kind_of Grocery::OnlineOrder
+			i.must_respond_to :customer
+			i.must_respond_to :status
+		}
+		
+		raw_orders = []
+		CSV.foreach('../support/online_orders.csv') { |row| #parses each row in the online_orders CSV
+			order_id = row[0]
+			customer = row[2]
+			status = row[3]
+			order_products = {}
+			row[1].split(';').each { |i| #like in orders.rb splits products by ;
+				single_item = i.split(':') #splits products into the name and cost
+				cost = single_item[1]
+				name = single_item[0]
+				order_products[name] = cost #adds the new prodcut to the hash with the correct value
+			}
+			
+			raw_orders << Grocery::OnlineOrder.new(order_id, order_products, customer, status) 
+		}
+		
+		online_orders.length.must_equal raw_orders.length
+	end
   end
   
   describe "OnlineOrder.find_by_customer" do
     it "Returns an array of online orders for a specific customer ID" do
-      # TODO: Your test code here!
-    end
+		a = Grocery::OnlineOrder.find_by_customer(1)
+		a.must_be_kind_of Grocery::Customer
```

Comment:
Based on the name of this test, shouldn't you be checking that `a` is of type `Array`, and the contents are instances of `Grocery::OnlineOrder`? If that's not what this method does, you should change the name of the test.